bigquery/v2/query/helper.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package query
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"cloud.google.com/go/bigquery/v2/apiv2/bigquerypb"
+	"cloud.google.com/go/bigquery/v2/apiv2_client"
+	"cloud.google.com/go/internal/uid"
+	"github.com/googleapis/gax-go/v2"
+	"google.golang.org/api/option"
+)
+
+// Helper for running queries in BigQuery. It is a lightweight wrapper
+// around the auto-generated BigQuery v2 client, focused on query operations.
+type Helper struct {
+	c         *apiv2_client.Client
+	projectID string
+}
+
+// NewHelper creates a new query helper. This helper should be reused instead of
+// created per-request.
+func NewHelper(c *apiv2_client.Client, projectID string, opts ...option.ClientOption) (*Helper, error) {
+	qc := &Helper{
+		c:         c,
+		projectID: projectID,
+	}
+	if qc.c == nil {
+		return nil, errors.New("missing bigquery client")
+	}
+	return qc, nil
+}
+
+// StartQuery executes a query using the stateless jobs.query RPC. It returns a
+// handle to the running query. The returned Query object can be used to wait for
+// completion and retrieve results.
+func (h *Helper) StartQuery(ctx context.Context, req *bigquerypb.PostQueryRequest, opts ...gax.CallOption) (*Query, error) {
+	qr := req.GetQueryRequest()
+	if qr == nil {
+		return nil, fmt.Errorf("bigquery: request is missing QueryRequest")
+	}
+	if qr.GetRequestId() == "" {
```

Comment:
If you're modifying the request, perhaps we should clone then modify.  Same feedback for the other methods below where we're monkeying with requests.